gateway/ln-gateway/src/state_machine/mod.rs

```diff
@@ -366,19 +352,14 @@ impl GatewayClientModule {
         fees: RoutingFees,
         lightning_context: LightningContext,
     ) -> anyhow::Result<()> {
-        {
-            let registration_info = self.to_gateway_registration_info(
-                route_hints,
-                time_to_live,
-                fees,
-                lightning_context,
-            );
+        let registration_info =
+            self.to_gateway_registration_info(route_hints, time_to_live, fees, lightning_context);
+        let gateway_id = registration_info.info.gateway_id;
 
-            let federation_id = self.client_ctx.get_config().global.federation_id();
-            self.register_with_federation_inner(federation_id, registration_info)
-                .await?;
-            Ok(())
-        }
+        let federation_id = self.client_ctx.get_config().global.federation_id();
+        self.module_api.register_gateway(&registration_info).await?;
+        debug!("Successfully registered gateway {gateway_id} with federation {federation_id}");
```

Comment:
nit: in the future we can use such refactors to move away from string interpolation to … whatever we call this syntax:

```rust
debug!(%gateway_id, %federation_id, "Successfully registered gateway with federation");
```